scanner/main.c

```diff
@@ -379,30 +380,26 @@ static void db_disc_handler(ble_db_discovery_evt_t * p_evt)
             // Rows are indexed by connection handles
             for(int i = 0; i < discovered_db->char_count; i++) 
             {
-                //NRF_LOG_INFO("Current connection handle is %d", m_conn_handle[p_evt->conn_handle]);
-                //NRF_LOG_INFO("uuid is %d", discovered_db->charateristics[i].characteristic.uuid.uuid);
-                //NRF_LOG_INFO("cccd handle is %d", discovered_db->charateristics[i].cccd_handle); 
-
                 characteristic[p_evt->conn_handle][i] = discovered_db->charateristics[i];

```

Comment:
I'm a bit worried about using the connection handle to index characteristic array directly. This assumes that connection handles will be in range [0 - NRF_SDH_BLE_CENTRAL_LINK_COUNT]. 
This might be true for the most part but can not be guaranteed as it is out of our control. 
E.g. it might work now but might change in a future SoftDevice implementation. 
Also, if a peripheral device disconnects and reconnects for any reason, it will get a new connection handle and the value will be outside of the range. Similar thing might also happen for some other reason that we didn't foresee.

So I suggest that you also keep an array of connection handles for all peripheral devices e.g.:
uint8_t peripheral_conn_handles[NRF_SDH_BLE_CENTRAL_LINK_COUNT]. Add the handles there sequentially as you connect to the peripherals.
Then if you later want to find the characteristic for a device with a given connection handle, first find the index of the handle in the peripheral_conn_handles array and use the same index to access the characteristic array.